### ISSUE TITLE
Take cookie banner component to accessibility clinic

### ISSUE THREAD

**Issue by kellylee-gds** [alphagov/govuk-frontend#2106]
## What
Take draft component to accessibility clinic.

## Why
To get expert feedback

## Who needs to know about this
Developers, designers

## Done when
- [x] Book session
- [x] Attended the session
- [x] Recorded feedback

**Comment by kellylee-gds** [comment]
Booked a slot 4 Feb but we plan to release the component before then. Are there any potential breaking changes that could be found which may make this approach problematic? @hannalaakso

**Comment by kellylee-gds** [comment]
Briefly spoke about this at stand up today. As we plan to do a 4.0 release soon (as we have prioritised making changes to the accordion component) any breaking changes that result from the accessibility clinic can be addressed then

**Comment by kellylee-gds** [comment]
Moved to blocked until we attend the clinic

**Comment by hannalaakso** [comment]
Doc of things we want to learn from taking the cookie banner to the accessibility clinic: https://docs.google.com/document/d/1esYkn7vK3Y3xy_Zi7071o5UES90ciN8z07FVSqLKXRU/edit#

**Comment by vanitabarrett** [comment]
Notes from the accessibility clinic:

Might not need to move focus AND use role=”alert” as role=”alert” is being applied dynamically rather than on page load. Hidden attribute / showing and hiding with JS is the exact use case for role alert. So we may not need to move focus. Could try and test with removing the focus. Although we also move focus as the button that originally had focus has now disappeared - so we move focus for 2 reasons.

Anika mentioned looking at 4.1.3 status messages. ‘…can be read / announced to user without receiving focus’. Think it mentions when you move focus, you don’t need to do alert stuff and vice versa - according to spec, might not be reality!

Unsure about placement of cookie banner before skip link – can see arguments either way.    If someone wants to ignore it and not interact, it might become an annoyance as they’ll have to go through it multiple times to get to page content / skip link.

No issues with Dragon 15.30 in IE11 or Dragon for client-side example.

Issues with hide on server side example in IE11 (issue with JS)

Gut feeling is not to move focus when a user clicks "Hide this message", but if we were to move the focus probably moving it to the ‘next thing’.

JAWS issue - If the button only gets read out, it might sound like it didn’t work. As the GOV.UK version works, this almost feels like a regression so we should probably look into it. One option might be to publish and fix later, but that might have implications for GOV.UK (regression in screenreader support and they’ll have to go back and fix it when we do).

**Comment by vanitabarrett** [comment]
**Actions:**

- Fix our full-page example which uses .remove() to remove the cookie banner from the DOM, which fails in IE
- Explore not moving focus to the confirmation banner (we're going to look at this as part of the JAWS bug anyway)
